hypothesis-python/tests/nocover/test_conjecture_engine.py

```diff
@@ -122,3 +124,21 @@ def f(data):
                 node = runner.tree[node][b]
             assert node in runner.dead
     assert len(seen) == 256
+
+
+def test_regression_1():
+    # This is a really hard to reproduce bug that previously triggered a very
+    # specific exception inside one of the shrink passes. It's unclear how
+    # useful this regression test really is, but nothing else caught the
+    # problem.
+    with deterministic_PRNG():
```

Comment:
I would integrate `deterministic_PRNG` into `run_to_buffer` - I can't think of a case where we *wouldn't* want this to be deterministic.  

Alternatively, perhaps the right place is actually in `ConjectureRunner._run`, here: 
https://github.com/HypothesisWorks/hypothesis/blob/b11921831fe4427ef8e38a955bb259cff943bb6c/hypothesis-python/src/hypothesis/internal/conjecture/engine.py#L769-L776